pkg/csi_driver/controller.go

```diff
@@ -101,15 +106,61 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 			return nil, status.Error(codes.AlreadyExists, err.Error())
 		}
 	} else {
+
+		// Get the reserved-IPV4-cidr from the request. It was not added in the service instance as it is only consumed by the controller
+		reservedIPV4CIDR, reservedIPV4CIDRPresent := req.GetParameters()[paramReservedIPV4CIDR]
+		if !reservedIPV4CIDRPresent {
+			return nil, fmt.Errorf("The required parameter reserved-ipv4-cidr was not provided")
+		}
+
+		// Prevents lock acquisition in case the CIDR is invalid
+		err = s.config.ipAllocator.ValidateCIDR(reservedIPV4CIDR)
```

Comment:
dont use `error` in place of true/false values. A validation should return true when validated and false when the cidr is invalid. It should only return error when there is a failure in the validation process itself

Reply:
Ok. Will change that.